robot_log_visualizer/ui/gui.py

```diff
@@ -367,17 +382,25 @@ def pauseButton_on_click(self):
                 video_item.media_player.pause()
 
         self.signal_provider.state = PeriodicThreadState.pause
-        # self.meshcat_provider.state = PeriodicThreadState.pause
+        self.meshcat_provider.state = PeriodicThreadState.pause
 
         self.logger.write_to_log("Dataset paused.")
 
     def plotTabCloseButton_on_click(self, index):
+        self.plottingLock.acquire()
         self.ui.tabPlotWidget.removeTab(index)
         self.plot_items[index].canvas.quit_animation()
+
+        # Update the indexes of plotData before deletion
+        for i in range(index, len(self.plotData.keys()) - 1):
+            self.plotData[i] = self.plotData[i + 1]
+        # Remove the last key
+        del self.plotData[list(self.plotData.keys())[-1]]
```

Comment:
can you explain why this is required? We can discuss this F2F

Reply:
@nicktrem can you check this comment?